ocaml/xapi/xapi_vm_helpers.ml

```diff
@@ -41,6 +41,45 @@ let compute_memory_overhead ~__context ~vm =
 
 let update_memory_overhead ~__context ~vm = Db.VM.set_memory_overhead ~__context ~self:vm ~value:(compute_memory_overhead ~__context ~vm)
 
+
+(* CP-18172: If we have nested virt in the platform map, we need a license *)
+(* It appears that xenguest checks the key with the following logic:
+   
+       s = xenstore_getsv(fmt, ap);
+       if (s) {
+           if (!strcasecmp(s, "true"))
+               value = 1;
+           else
+           {
+               errno = 0;
+               value = strtoull(s, NULL, 0);
+               if ( errno )
+                   value = 0;
+           }
+           free(s);
+       }
+
+   so we do something equivalent here *)
+let check_restricted_platform_flags ~__context platform =
+  (* Takes a key and a predicate (string -> bool). If the key has an
+     associated value in the platform map, and the predicate is true
+     for that value, assert that the feature is enabled. *)
+  let check_key key predicate f =
+    if List.mem_assoc key platform && predicate (List.assoc key platform)
+    then Pool_features.assert_enabled ~__context ~f
+  in
+
+  let enabled_in_xenguest v =
```

Comment:
I think this code should be accessible from the outside to make sure we can use it in the VM lifecycle checks where currently there is a different implementation.